postgresql/sql-scripts/citydb-pkg/delete.sql

```diff
@@ -146,6 +150,129 @@ END;
 $body$
 LANGUAGE plpgsql STRICT;
 
+/******************************************************************
+* terminate feature based on an id array
+******************************************************************/
+CREATE OR REPLACE FUNCTION citydb_pkg.terminate_feature(pid_array bigint[], metadata JSON DEFAULT '{}') RETURNS SETOF BIGINT AS
+$body$
+DECLARE
+  terminated_ids bigint[] := '{}';
+  child_feature_ids bigint[] := '{}';
+  termination_timestamp TIMESTAMP;
+  update_reason TEXT;
+  update_person TEXT DEFAULT USER;
+  udpate_lineage TEXT;
+BEGIN
+  termination_timestamp = now();
+
+  IF metadata IS NOT NULL THEN
+    IF metadata->>'reason_for_update' IS NOT NULL THEN
+      update_reason = metadata->>'reason_for_update';
+    END IF;
+
+    IF metadata->>'updating_person' IS NOT NULL THEN
+      update_person = metadata->>'updating_person';
+    END IF;
+
+    IF metadata->>'lineage' IS NOT NULL THEN
+      udpate_lineage = metadata->>'lineage';
+    END IF;
+  END IF;
```

Comment:
I suggest deleting lines 166-180 and checking for both NULL and default values directly in the UPDATE statement using COALESCE (see comment below).

Reply:
Done with f33a79e1